### PR TITLE
Fix nav and footer

### PR DESCRIPTION
**Pull request by marleyyvon**
- Added "My Profile" and "Settings" options to the dropdown menu with a custom redirect to "Goodbye" on sign-out.
- Created new routes with placeholder pages for "My Profile," "Settings," and "Goodbye."
- Removed footer copyright and updated styling to match the header.
- Added placeholder pages for "Goodbye," "About," and "Privacy Policy" with basic routing.
- Fixed footer positioning across all pages for consistent display.
- Adjusted the size of `LoginCard` for better alignment and visual consistency.
- Created separate pages for each user role (admin, faculty, student, super_admin) and added role-based redirection on the homepage.
- Added a security guard to restrict access to admin pages, allowing only authorized users to view them.
- Updated the Navbar to conditionally show navigation links only for logged-in users, hiding them from unauthenticated users to improve UI-level security.

### CODE REVIEW

**Review by miaadele — APPROVED** [review]

**Review by vaniaggarwal — APPROVED** [review]

File: src/components/AuthGuard.jsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+"use client";
+
+import { useSession } from "next-auth/react";
+import { useRouter } from "next/navigation";
+import PropTypes from "prop-types";
+import LoginCard from "@/components/LoginCard";
```
Comment (v3ceban) [inline src/components/AuthGuard.jsx:6]: This can also be a server component

File: src/lib/server/getServerSession.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { auth } from "@/lib/auth";
+
+export async function getServerSession() {
+  return await auth();
+}
```
Comment (v3ceban) [inline src/lib/server/getServerSession.js:5]: What is the point of creating this function if it returns the result of **@/lib/auth**?

File: src/components/AuthGuard.jsx (hunk 1 above)
Comment (v3ceban) [reply src/components/AuthGuard.jsx:6]: Also, didn't we already have protected routes with route groups?

File: src/components/AuthGuard.jsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+};
+
+AuthGuard.propTypes = {
+  requiredRole: PropTypes.oneOf(["student", "faculty", "admin", "super_admin", "any"]),
+  children: PropTypes.node.isRequired,
+};
+
+AuthGuard.defaultProps = {
+  requiredRole: "any", // "any" allows any logged-in user to access the page
+};
```
Comment (v3ceban) [inline src/components/AuthGuard.jsx:35]: I'm not sure if this works... If it does - good, but a more orthodox way of assigning requiredRole to a default of "any" would be:
const AuthGuard = ({requiredRole = "any", children}) => { ... }

File: src/components/Navbar/Navbar.jsx [hunk 4]
```diff
@@ -4,34 +4,39 @@ import React, { useState } from "react";
...
   const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+  const router = useRouter();
 
   const toggleDropdown = () => setIsDropdownOpen((prev) => !prev);
 
+  const handleSignOut = async () => {
+    await signOut({ redirect: false });
+    router.push("/goodbye");
+  };
+
```
Comment (v3ceban) [inline src/components/Navbar/Navbar.jsx:20]: This will conflict with the changes in #22
This whole Navbar should be a server component.

**Review by v3ceban — CHANGES_REQUESTED** [review]

File: src/app/Home.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+          router.push("/student");
+          break;
+        case "super_admin":
+          router.push("/super-admin");
+          break;
+        default:
+          break;
+      }
+    }
+  }, [user, router]);
```
[the remark this answers is not in the document]
Comment (marleyyvon) [reply src/app/Home.jsx:30]: Updated

File: src/components/AuthGuard.jsx (hunk 1 above)
Comment (marleyyvon) [reply src/components/AuthGuard.jsx:6]: Converted AuthGuard.jsx to a server component using auth() for server-side role checks, removing useSession and useRouter. Verified route groups are protected, but kept AuthGuard for now.

File: src/lib/server/getServerSession.js (hunk 2 above)
Comment (marleyyvon) [reply src/lib/server/getServerSession.js:5]: Refactored session handling: removed `getServerSession` function and replaced it with direct `auth()` call in `page.jsx`. Cleaned up imports to simplify session retrieval. Failed to recognize we already have auth direction. Updated!

File: src/components/Navbar/Navbar.jsx (hunk 4 above)
Comment (marleyyvon) [reply src/components/Navbar/Navbar.jsx:20]: Will make a note to update this in next sprint or after the merge.

**Pull request merged**